x/stake/keeper/validator.go

```diff
@@ -246,13 +252,64 @@ func (k Keeper) UpdateValidator(ctx sdk.Context, validator types.Validator) type
 			bz := k.cdc.MustMarshalBinary(validator.ABCIValidator())
 			store.Set(GetTendermintUpdatesKey(validator.Owner), bz)
 		}
-
 	}
 
 	k.SetValidator(ctx, validator)
 	return validator
 }
 
+// updateCliffValidator determines if the current cliff validator needs to be
+// updated or swapped.
+func (k Keeper) updateCliffValidator(ctx sdk.Context, affectedVal types.Validator) {
+	var cliffValByPower types.Validator
+
+	cliffAddr := sdk.AccAddress(k.GetCliffValidator(ctx))
+	if !bytes.Equal(cliffAddr, affectedVal.Owner) {
+		return
+	}
+
+	store := ctx.KVStore(k.storeKey)
+	pool := k.GetPool(ctx)
+
+	// NOTE: We get the power via affectedVal since the store (by power key)
+	// has yet to be updated.
+	affectedValPower := affectedVal.GetPower()
+
+	// Create a validator iterator ranging from smallest to largest in power
+	// where only the smallest by power is needed.
+	iterator := sdk.KVStorePrefixIterator(store, ValidatorsByPowerIndexKey)
+	offset := 0
+	for {
+		if !iterator.Valid() || offset == 1 {
```

Comment:
I don't understand this for loop. Why are we breaking when `offset == 1`. It seems to me that the only code path here is offset incrementing after the first iteration, and this for loop breaking. (Which means it shouldn't be a loop)

Reply:
I originally had it with an offset of two. In which case I needed the loop, but then discovered that would not work. I glossed over the fact that I no longer need the loop.